MetaU Capstone Project/API/SpotifyManager.m

```diff
@@ -247,6 +260,144 @@ - (void)searchTrack:(NSString *)query type:(NSString *)type result:(void (^)(NSD
         NSData *dataUTF8 = [strISOLatin dataUsingEncoding:NSUTF8StringEncoding];
         dataRecieved = [NSJSONSerialization JSONObjectWithData:dataUTF8 options:0 error:&error];
         
+        if (dataRecieved != nil) {
+            parsingFinished([dataRecieved copy]);
+        } else {
+            NSLog(@"Error: %@", error.localizedDescription);
+            parsingFinished([[NSDictionary alloc] init]);
+        }
+    }] resume];
+}
+
+- (void)addQueueToSpotify: (NSString *)trackURI {
+    [[self.appRemote playerAPI] enqueueTrackUri:trackURI callback:^(id  _Nullable result, NSError * _Nullable error) {
+            if (error != nil) {
+                NSLog(@"Error: %@", error.localizedDescription);
+            } else {
+                NSLog(@"%@ added to Spotify queue", trackURI);
+            }
+    }];
+}
+
+- (NSDictionary *)currentTrackInfo: (id<SPTAppRemoteTrack>)track {
+    NSDictionary *trackDetails = @{
+            @"name" : track.name,
+            @"URI" : track.URI,
+            @"contextURI" : track.album.URI,
+            @"artist" : track.artist.name,
+            @"album" : track.album.name,
+            @"images" : @[]
+    };
+
+    return (NSDictionary *)trackDetails;
```

Comment:
You don't need to cast it to an `NSDictionary` here. You've already defined the type of `trackDetails` on line 283

Sooo just:

```
return trackDetails;
```